overlord/devicestate/helpers.go

```diff
@@ -45,3 +50,17 @@ func gadgetDataFromInfo(info *snap.Info, model *asserts.Model) (*gadget.GadgetDa
 	}
 	return &gadget.GadgetData{Info: gi, RootDir: info.MountDir()}, nil
 }
+
+func maybeGetSystemForPreseeding() (label string, err error) {
+	systemLabels, err := filepath.Glob(filepath.Join(dirs.SnapSeedDir, "systems", "*"))
+	if err != nil && !os.IsNotExist(err) {
+		return "", fmt.Errorf("cannot list available systems: %v", err)
+	}
+	if len(systemLabels) == 0 {
+		return "", nil
```

Comment:
why doesn't this fail in this case?

Reply:
You're right, it should. Annoyingly making it fail affects a bunch of existing tests in unfortunate ways; for no good reason firstbootPreseed16Suite reuses firstBootBaseTest/firstBoot16BaseTest which have assumptions about running on core (release.MockOnClassic(false); also creating devicemgr early with this assumption is a problem), this wasn't a factor for partial and selective preseed tests for classic, but bites now. I see no easy way to untangle this, it should probably be attacked in a separate PR 😞

Reply:
That's done now.